### PR TITLE
doc: add Envoy Gateway exported metrics

### LINKED ISSUE #3404: docs: Observability of Envoy Gateway Component (not Envoy)
*Observability of Envoy Gateway Component (not Envoy)*:

To have full EG system observability and ease troubleshooting of various failure modes, is there is EG native metrics (not Envoy proxy) that helps to have insights into EG subcomponents like:

* Total/failed to translate resources (Resource translator)
* Total/failed to translate xDS resources (xDS translator)
* Total/failed to translate Infra resources (Infra translator)

Introducing such metrics (if they are not yet supported) will ease EG operators to troubleshoot various failure scenarios and increase confidence for prod deployment.

Could not find mentions of EG native metric in user docs.

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3404

### CODE REVIEW

**Review by Xunzhuo — APPROVED** [review]
LGTM, thanks for adding it!

File: site/content/en/latest/tasks/observability/gateway-exported-metrics.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Envoy Gateway collects the following metrics in Watching Components:
+
+| Name                                    | Description                                            |
+|-----------------------------------------|--------------------------------------------------------|
+| `watchable_depth`                       | Current depth of watchable map.                        |
+| `watchable_subscribed_duration_seconds` | How long in seconds a subscribed watchable is handled. |
+| `watchable_subscribed_total`            | Total number of subscribed watchable.                  |
+| `watchable_subscribed_errors_total`     | Total number of subscribed watchable errors.           |
+
+Each metrics includes the `runner` label to identify the corresponding components,
```
Comment (cnvergence) [inline site/content/en/latest/tasks/observability/gateway-exported-metrics.md:27]:
```suggestion
Each metric includes the `runner` label to identify the corresponding components,
```
small nit

**Review by cnvergence — APPROVED** [review]

File: site/content/en/latest/tasks/observability/gateway-exported-metrics.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+title: "Gateway Exported Metrics"
```
Comment (arkodg) [inline site/content/en/latest/tasks/observability/gateway-exported-metrics.md:2]: can we call this Envoy Gateway Metrics or Control Plane Metrics ?

Comment (shawnh2) [reply site/content/en/latest/tasks/observability/gateway-exported-metrics.md:2]: How about `Envoy Gateway Metrics List`? The `Envoy Gateway Metrics` or `Control Plane Metrics` may be confusing, we need a name to tell whether this doc is for usage or for metrics list.

Comment (arkodg) [reply site/content/en/latest/tasks/observability/gateway-exported-metrics.md:2]: sure that works too, merging this PR for now

**Pull request merged**